t21log/src/main/java/com/tempos21/android/commons/utils/FileLogger.java

```diff
@@ -0,0 +1,134 @@
+package com.tempos21.android.commons.utils;
+
+import android.content.Context;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Locale;
+
+/**
+ * Class that writes log messages to a logFile.
+ */
+class FileLogger extends Logger {
+
+    private static final String T21_LOG_TAG = "T21Log";
+
+    private static final String SLASH = "/";
+
+    private static final String COLON = ":";
+
+    private static final String DATE_TIME_FORMAT = "yyyy/MM/dd HH:mm:ss.SSS";
+
+    private static final String LINE_SEPARATOR_PROPERTY = "line.separator";
+
+    private static final String LOG_PATH = "logs";
+
+    private static final String LOG_FILE_NAME = "T21Log.log";
+
+    private static File logFile;
+
+    private static boolean ENABLED = false;
+
+    /**
+     * Initialize the FileLogger to write to a file.
+     *
+     * @param tag:     Your appName or something you want to appear in log
+     * @param enabled: If you want to enable or disable the log
+     * @param context: Android app context
+     */
+    static void initialize(String tag, boolean enabled, Context context) {
+        Logger.initialize(tag);
+        ENABLED = enabled;
+        File logsPath = new File(context.getFilesDir(), LOG_PATH);
+        if (logsPath.exists() || (!logsPath.exists() && logsPath.mkdir())) {
+            logFile = new File(logsPath, LOG_FILE_NAME);
+
+            try {
+                if (!logFile.exists()) {
+                    logFile.createNewFile();
+                }
+            } catch (IOException e) {
+                ENABLED = false;
+                android.util.Log
+                        .e(T21_LOG_TAG, "Unable to create log file '" + logFile.getName() + "'. Logging to file disabled.\n" + e);
+            }
+        } else {
+            ENABLED = false;
+            android.util.Log.e(T21_LOG_TAG, "Unable to create 'logs' directory. Logging to file disabled");
+        }
+    }
+
+    static File getLogFile() {
+        return logFile;
+    }
+
+    private static void writeToFile(LogLevel logLevel, String logTag, String message) {
```

Comment:
I think there is a better way to catch the exceptions for the i/o streams (try-catch-finally), to be sure that streams are closed in case of errors. Something like this (just an example):
```java
ObjectOutputStream oos = null;
try {
   oos = new ObjectOutputStream(new FileOutputStream(file));
   oos.writeObject(shapes);
   oos.flush();
} catch (FileNotFoundException ex) {
    // complain to user
} catch (IOException ex) {
    // notify user
} finally {
    if (oos != null) {
        try {
            oos.close();
        } catch (IOException ex) {
            // ignore ... any significant errors should already have been
            // reported via an IOException from the final flush.
        }
    }
}
```